src/model/model_eval.c

```diff
@@ -694,10 +708,23 @@ static value_t eval_arith_pprod_algebraic(evaluator_t *eval, pprod_t *p) {
 }
 #endif
 
+/*
+ * Finite field atom: t == 0
+ */
+static value_t eval_arith_ff_eq(evaluator_t *eval, term_t t) {
+  value_t v;
+
+  v = eval_term(eval, t);
+  assert(object_is_finitefield(eval->vtbl, v));
+
+  value_ff_t *v_ff = vtbl_finitefield(eval->vtbl, v);
+  return vtbl_mk_bool(eval->vtbl, q_is_zero(&v_ff->value));
+}
+
 /*
  * Power product: arithmetic
  */
-static value_t eval_arith_pprod(evaluator_t *eval, pprod_t *p) {
+static value_t eval_arith_pprod(evaluator_t *eval, pprod_t *p, const rational_t *mod) {
```

Comment:
please add a comment about what mod means here

Reply:
I'll split the code in `eval_arith_pprod` and `eval_arith_ff_pprod`